src/main/webapp/assets/modules/talent-search/company-profile/company-profile.con.js

```diff
@@ -1,30 +1,77 @@
-techlooper.controller("companyProfileController", function ($scope, companyProfileService, $routeParams, $http, jsonValue, $timeout) {
+techlooper.controller("companyProfileController", function ($scope, $sce, companyProfileService, $routeParams,
+                                                            $http, jsonValue, $timeout, utils, $rootScope, $location) {
   $scope.follow = companyProfileService.followManager;
 
   var companyName = $routeParams.companyName;
-  $http.get(jsonValue.httpUri.company + "/" + companyName).success(function(data) {
-    $.each(data.benefits, function(i, benefit) {
-      benefit.name = jsonValue.benefits[benefit.benefitId].name;
-      benefit.iconName = jsonValue.benefits[benefit.benefitId].iconName;
-    });
-    $.each(data.industries, function(i, industry) {
-      industry.value = jsonValue.industries[industry.industryId].value;
-    });
+  var parts = companyName.split("-");
+  if (parts.length < 3) {
+    $location.path("/");
+    return;
+  }
+
+  if (parts[parts.length - 1] === "cp") {
+    var companyId = parts[parts.length - 2];
+  }
+
+  if (companyId === undefined) {
+    $location.path("/");
+    return;
+  }
+
+  if (utils.hasNonAsciiChar(companyName)) {
+    $location.path("/companies/" +utils.toAscii(companyName));
+    return;
+  }
+
+  $http.get(jsonValue.httpUri.companyId + "/" + companyId)
+    .success(function (data) {
+      $.each(data.benefits, function (i, benefit) {
+        benefit.name = jsonValue.benefits[benefit.benefitId].name;
+        benefit.iconName = jsonValue.benefits[benefit.benefitId].iconName;
+      });
+      $.each(data.industries, function (i, industry) {
+        industry.value = jsonValue.industries[industry.industryId].value;
+      });
+
+      data.totalViews = 0;
+      data.totalApplications = 0;
+      $.each(data.jobs, function (i, job) {
+        data.totalViews += job.numOfViews;
+        data.totalApplications += job.numOfApplications;
+      });
+      data.totalViews = data.totalViews.toLocaleString();
+      data.totalApplications = data.totalApplications.toLocaleString();
+      data.companySize = jsonValue.companySizes[data.companySizeId];
+      $scope.companyInfo = data;
+      console.log($scope.companyInfo)
+      if(data.jobVideoURLs != null){
+        var fullURL = data.jobVideoURLs[0];
+      }else{
+        var fullURL = '';
```

Comment:
'fullURL' is already defined.<br>Mixed double and single quotes.<br>'fullURL' is defined but never used.